### PR TITLE
Add static routes

### PR DESCRIPTION
**Pull request by yiiliveext**
| Q             | A
| ------------- | ---
| Is bugfix?    | ❌
| New feature?  | ✔️
| Breaks BC?    | ❌
| Fixed issues  | #100

### CODE REVIEW

**Comment by Mister-42** [comment]
Wouldn't Route::static make more sense?

**Comment by yiiliveext** [comment]
> Wouldn't Route::static make more sense?

What do you mean? 
The example from your issue
```php
//Assets for urlGenerator, requests will not be handled by Yii
    Route::get('/images/{file}')
        ->static() 
        ->host($appParams->assetDomain)
        ->name('img'),
```
That's all. You can use static routes in a common routes file/array.

**Comment by Mister-42** [comment]
I meant 

```php
    //Assets for urlGenerator, requests will not be handled by Yii
    Route::static('/images/{file}')
        ->host($appParams->assetDomain)
        ->name('img'),
```

Sorry, I commented without enough information.

**Comment by yiiliveext** [comment]
> I meant
> 
> ```
>     //Assets for urlGenerator, requests will not be handled by Yii
>     Route::static('/images/{file}')
>         ->host($appParams->assetDomain)
>         ->name('img'),
> ```
> 
> Sorry, I commented without enough information.

Maybe it makes sense. Need to review from more devs.

**Comment by samdark** [comment]
There was an ability to make URL rule with matching only or generation only in Yii 2: https://github.com/yiisoft/yii2/blob/master/framework/web/UrlRule.php#L38

I haven't personally used it myself so I'm not sure about use cases.

**Comment by yiiliveext** [comment]
> I meant
> 
> ```
>     //Assets for urlGenerator, requests will not be handled by Yii
>     Route::static('/images/{file}')
>         ->host($appParams->assetDomain)
>         ->name('img'),
> ```

Done

File: tests/RouteTest.php [hunk 1]
```diff
@@ -124,6 +124,47 @@ public function testDefaults(): void
...
+        Route::static('/')->middleware(fn () => 1);
+    }
+
+    public function testStaticFailWihPrependMiddleware(): void
+    {
+        $this->expectExceptionMessage("Static route can not has middleware.");
+        Route::static('/')->middleware(fn () => 1);
+    }
+
+    public function testStaticFailWihAddAction(): void
```
Comment (samdark) [inline tests/RouteTest.php:162]:
```suggestion
    public function testStaticFailWithAddAction(): void
```

File: src/Route.php [hunk 2]
```diff
@@ -139,6 +146,19 @@ public static function options(string $pattern, ?MiddlewareDispatcher $dispatche
         return self::methods([Method::OPTIONS], $pattern, $dispatcher);
     }
 
+    /**
+     * @param string $pattern
+     *
+     * @return self
+     */
+    public static function static(string $pattern): self
```
Comment (samdark) [inline src/Route.php:154]: There is no need for a separate method. Generation-only route is the one that doesn't have action.

**Comment by samdark** [comment]
👍

**Pull request merged**